composer/checkpoint/state_dict.py

```diff
@@ -158,6 +169,82 @@ def _get_model_state_dict_with_fsdp_context_manager(model: nn.Module, sharded_st
     return model_state_dict
 
 
+def get_resumption_state_dict(state: State) -> Dict[str, Any]:
+    """Generate the state dict for any objects needed for resumption.
+
+    This includes:
+        * timestamp
+        * scheduler
+        * dataset_state
+        * scaler
+        * rank_zero_seed
+        * callbacks
+        * algorithms
+
+    Returns:
+        The state dict containing the objects needed for resumption.
+    """
+    resumption_state_dict = {}
+    resumption_state_dict['dataset_state'] = get_dataset_state_dict(
+        state.train_dataloader,
+        state.timestamp,
+    )
+    resumption_state_dict['timestamp'] = state.timestamp.state_dict()
+
+    scheduler_state_dict = _make_state_dict_for_list_of_objects(state.schedulers)
+    if scheduler_state_dict != {}:
+        resumption_state_dict['schedulers'] = scheduler_state_dict
+
+    callbacks_state_dict = _make_state_dict_for_list_of_objects(state.callbacks)
+    if callbacks_state_dict != {}:
+        resumption_state_dict['callbacks'] = callbacks_state_dict
+
+    algorithms_state_dict = _make_state_dict_for_list_of_objects(state.algorithms)
+    if algorithms_state_dict != {}:
+        resumption_state_dict['algorithms'] = algorithms_state_dict
+
+    if state.scaler is not None:
+        scaler_sd = _make_state_dict_for_list_of_objects(state.scaler)
+        if scaler_sd != {}:
+            resumption_state_dict['scaler'] = state.scaler.state_dict()
+
+    resumption_state_dict['rank_zero_seed'] = state.rank_zero_seed
+    resumption_state_dict['run_name'] = state.run_name
+
+    return resumption_state_dict
+
+
+def _make_state_dict_for_list_of_objects(objects: Union[Sequence[Any], Any]) -> Dict[str, Any]:
+    object_dict = {}
+    if not isinstance(objects, Sequence):
+        objects = [objects]
+    for obj in objects:
+        if not hasattr(obj, 'state_dict') or obj.state_dict() == {}:
+            continue
+        object_dict[type(obj).__qualname__] = obj.state_dict()
```

Comment:
If there are more than 1 `obj`s in the `objects`,  will the `type(obj).__qualname__` be the same and overwrite the previous one ?

Reply:
oh interesting edge case. seems rare, but we could use a list perhaps?

Reply:
I don't have full context here. I think as long as it's consistent for save and load, we should be good ?

Reply:
ok yeah It should be good now